src/app/utils/bibDetailsUtils.jsx

```diff
@@ -212,6 +212,73 @@ const constructSubjectHeadingsArray = (url = '') => {
     });
 };
 
+/**
+ * isRtl(string)
+ * Returns true (or false) depending on whether the string needs to be read right to left
+ * @param {string} string
+ * @return  {boolean}
+ */
+const isRtl  = (string) => {
+  if (typeof string !== 'string') { return null }
+  return string.substring(0, 1) === '\u200F'
+}
+
+/**
+ * stringDirection(string)
+ * 'rtl' if the string needs to be read right to left, otherwise null
+ * @param {string} string
+ * @return {string}
+ */
+const stringDirection = (string) => {
+  return isRtl(string) ? 'rtl' : null
+}
+
+/**
+ * combineMatching(el1, el2)
+ * Combines properties from matching (i.e. parallel) elements as necessary
+ * Right now, this is only needed to add the 'noteType' in case of parallel notes
+ * @param {string} el1
+ * @param {object} el2
+ * @return {object}
+ */
+
+ const combineMatching = (el1, el2) => (el2 && el2.noteType) ? { noteType: el2.noteType, '@type' : el2['@type'], prefLabel: el1 } : el1;
+
+
+/**
+ * interleave(arr1, arr2)
+ * Given two arrays, returns the elements interleaved, with falsey elements removed.
+ * Example: interleave ([1, 2, null, 3], [5,6,7,8,9]) =>
+ * [1,5,2,6,7,3,8,9].
+ * Assumes that arr2 is at least as long as arr1.
+ *
+ * @param {array} arr1
+ * @param {array} arr2
+ * @return {array}
+ */
+const interleave = (arr1, arr2) => arr2.reduce((acc, el, id) => (arr1[id] && acc.push(combineMatching(arr1[id], el)), el && acc.push(el), acc), [])
```

Comment:
Nitpick: This seems fine but it'd be easier to read with some line breaks.

Reply:
Good point, I've broken it up some and re-written for readability